torchvision/csrc/cpu/ROIAlign_cpu.cpp

```diff
@@ -144,14 +144,7 @@ void ROIAlignForward(
     // Force malformed ROIs to be 1x1
     T roi_width = std::max(roi_end_w - roi_start_w, (T)1.);
     T roi_height = std::max(roi_end_h - roi_start_h, (T)1.);
-    if (aligned) {
-      AT_ASSERTM(
-          roi_width >= 0 && roi_height >= 0,
-          "ROIs in ROIAlign do not have non-negative size!");
-    } else { // for backward-compatibility only
-      roi_width = std::max(roi_width, (T)1.);
-      roi_height = std::max(roi_height, (T)1.);
-    }
```

Comment:
although the assertion branch does not matter (from the discussion in thread https://github.com/pytorch/vision/pull/1908/commits/239c6197c93af491f543f4d8df99923b55eaeed8#r384267434), the else branch is important and makes the behavior different from detectron2. Now width/height are always clipped to 1.0, but in detectron2 they are clipped to 1.0 only when aligned=False.

Reply:
Thanks for the catch, I'll prepare a patch for this and backport to the 0.7 branch so that it gets to the release